src/lib/interceptor.ts

```diff
@@ -240,6 +246,32 @@ export class ResponseParsingInterceptor {
   }
 }
 
+export class PreventMutationsInterceptor {
+  private requestInterceptor: grpc.Requester;
+  private blankInterceptor: grpc.Requester;
+
+  constructor() {
+    this.requestInterceptor = this.buildRequester();
+    this.blankInterceptor = buildBlankInterceptor();
+  }
+
+  public intercept(options: grpc.CallOptions, nextCall: NextCall): grpc.InterceptingCall {
+    const isMutation = isMutationRequest(options);
```

Comment:
Does this need to be put into a variable?

Reply:
I prefer this style, but I can change it, sure.